### PR TITLE
Add documentation to help with customisation of `contrib.modeladmin`

### PR DESCRIPTION
**Pull request by ababic**

### CODE REVIEW

File: docs/reference/contrib/modeladmin/chooseparentview.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+======================================
+Customising ``ChooseParentView``
+======================================
+
+When adding adding new page via Wagtail's explorer view, you
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/chooseparentview.rst]: Duplicated "adding".

File: docs/reference/contrib/modeladmin/chooseparentview.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+essentially choose where you want to add a new page by navigating the
+relevant part of the page tree and choosing to 'add a child page' to your 
+chosen parent page. Wagtail then asks you to select what type of page you'd
+like to add.
+
+When adding a page from a ``ModelAdmin`` list page, we know what type of page
+needs to be added, but we might not automatically know where in the page tree
+it should be added. If there's only one possile choice of parent for a new page
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/chooseparentview.rst]: "possible"

File: docs/reference/contrib/modeladmin/indexview.rst [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+---------------------------
+
+**Expected value**: A list or tuple, where each item is the name of a model field
+of type ``CharField``, ``TextField``, ``RichTextField`` or ``StreamField``.
+
+Set ``search_fields`` to enable a search box at the top of the index page
+for your model. You should add names of any fields on the model that should 
+be searched whenever somebody submits a search query using the search box.
+
+Searching is all handled via Django's queryset API, rather than using the
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/indexview.rst]: Unnecessary "the"?

File: docs/reference/contrib/modeladmin/indexview.rst [hunk 4]
```diff
@@ -0,0 +1,590 @@
...
+
+**Must return**: A list
+
+The ``get_extra_class_names_for_field_col`` method allows you to add additional
+CSS class names to any of the columns defined by ``list_display`` for your
+model. The method takes two parameters:
+
+- 	``obj``: the object being represented by the current row
+-	``field_name``: the item from ``list_display`` being represented by the
+	curent column
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/indexview.rst]: "current"

File: docs/reference/contrib/modeladmin/indexview.rst [hunk 5]
```diff
@@ -0,0 +1,590 @@
...
+**Must return**: A dictionary
+
+The ``get_extra_attrs_for_field_col`` method allows you to add additional HTML
+attributes to any of the columns defined in ``list_display``. Like the
+``get_extra_class_names_for_field_col`` method above, this method takes two
+parameters: 
+
+- 	``obj``: the object being represented by the current row
+-	``field_name``: the item from ``list_display`` being represented by the
+	curent column
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/indexview.rst]: "current"

File: docs/reference/contrib/modeladmin/indexview.rst [hunk 6]
```diff
@@ -0,0 +1,590 @@
...
+        return attrs
+```
+
+.. _modeladmin_thumbnailmixin:
+
+----------------------------------------------------
+``wagtal.contrib.modeladmin.options.ThumbnailMixin``
+----------------------------------------------------
+
+If you're using ``wagtailimages.Image`` to define an image for each items in
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/indexview.rst]: "item"

File: docs/reference/contrib/modeladmin/indexview.rst [hunk 7]
```diff
@@ -0,0 +1,590 @@
...
+your model, ``ThumbnailMixin`` can help you add thumbnail versions of that 
+image to each row in ``IndexView``. To use it, simply extend ``ThumbnailMixin``
+as an well as ``ModelAdmin`` when defining your ``ModelAdmin`` class, and
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/indexview.rst]: Remove "an"

File: docs/reference/contrib/modeladmin/inspectview.rst [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+**Expected value:** A list or tuple, where each item is the name of a field 
+that you'd like ``InpectView`` to render.
+
+A sensible value will be rendered for most field types.
+
+If a field happens to be a ``ForeignKey`` linking to the 
+``wagtailimages.Image`` Model, a thumbnail of that image will be rendered. 
+
+If a field happens to be a ``ForeignKey`` linking to the
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/inspectview.rst]: "model"

File: docs/reference/contrib/modeladmin/inspectview.rst [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+.. _modeladmin_inspect_view_fields_exclude:
+
+------------------------------------------
+``ModelAdmin.inspect_view_fields_exclude``
+------------------------------------------
+
+**Expected value:** A list or tuple, where each item is the name of a field 
+that you'd like to exclude from ``InpectView``
+
+**Note:** If both ``inspect_view_fields`` and ``insepct_view_fields_exclude``
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/inspectview.rst]: "inspect_view_fields_exclude"

File: docs/reference/contrib/modeladmin/menu_item.rst [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+item that appears in Wagtail's sidebar. If not set, the menu item will use
+``verbose_name_plural`` from your model's ``Meta`` data.
+
+.. _modeladmin_menu_icon:
+
+-------------------------
+``ModelAdmin.menu_icon``
+-------------------------
+
+**Expected value**: A string matching one Wagtail's icon class names.
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/menu_item.rst]: Missing "of".

File: docs/reference/contrib/modeladmin/menu_item.rst [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+If you want to change the icon used to represent your Model, you can set the
+``menu_icon`` attribute on your class to use one of the other icons available
+in Wagtail's CMS. The same icon will be used for the menu item in Wagtail's
+sidebar, and will also appear in the header on the list page and other views
+for your Model. If not set, ``'doc-full-inverse'`` will be used for
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/menu_item.rst]: model"

File: docs/reference/contrib/modeladmin/primer.rst [hunk 12]
```diff
@@ -0,0 +1,397 @@
+===================================
+``modeladmin`` customisation primer
+===================================
+
+The ``modeladmin`` app is designed to offer you as much flexibility as possible 
+in how your model and it's objects are represented in Wagtail's CMS. It's
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/primer.rst]: "its"

File: docs/reference/contrib/modeladmin/primer.rst [hunk 13]
```diff
@@ -0,0 +1,397 @@
...
+
+-----------------------------------------------
+Adding additional stylesheets and/or javascript
+-----------------------------------------------
+
+The ``ModelAdmin`` class provides several attributes to enable you to easily
+add additional stylesheets and javascript to the admin interface for your
+model. Each atttribute simply needs to be a list of paths to the files you
+want to include. If the path is for a file in your project's static directory,
+wagtail will automatically prepended paths for each path with ``STATIC_URL``,
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/primer.rst]: "Wagtail"

File: docs/reference/contrib/modeladmin/primer.rst [hunk 14]
```diff
@@ -0,0 +1,397 @@
...
+so you don't need to repeat it each time in your list of paths.
+
+If you'd like to add styles or scripts to the ``IndexView``, you should set the
+following attributes:
+
+-	``index_view_extra_css`` -  Where each item is the path name of a
+	pre-compliled stylesheet that you'd like to include.
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/primer.rst]: "compiled"

File: docs/reference/contrib/modeladmin/primer.rst [hunk 15]
```diff
@@ -0,0 +1,397 @@
...
+
+-	``index_view_extra_js`` - Where each item is the path name of a javascipt
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/primer.rst]: "javascript"

File: docs/reference/contrib/modeladmin/primer.rst [hunk 16]
```diff
@@ -0,0 +1,397 @@
...
+	file that you'd like to include.
+
+If you'd like to do the same for `CreateView` and ``EditView``, you should set
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/primer.rst]: ``CreateView``

File: docs/reference/contrib/modeladmin/primer.rst [hunk 17]
```diff
@@ -0,0 +1,397 @@
...
+the following attributes:
+
+-	``form_view_extra_css`` -  Where each item is the path name of a
+	pre-compliled stylesheet that you'd like to include.
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/primer.rst]: "compiled"

File: docs/reference/contrib/modeladmin/primer.rst [hunk 18]
```diff
@@ -0,0 +1,397 @@
...
+
+-	``form_view_extra_js`` - Where each item is the path name of a javascipt
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/primer.rst]: "javascript"

File: docs/reference/contrib/modeladmin/primer.rst [hunk 19]
```diff
@@ -0,0 +1,397 @@
...
+	file that you'd like to include.
+
+And if you're using the ``InspectView`` for your model, and want to do the same
+for that view, your should set the following attributes:
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/primer.rst]: "you"

File: docs/reference/contrib/modeladmin/primer.rst [hunk 20]
```diff
@@ -0,0 +1,397 @@
...
+
+-	``inspect_view_extra_css`` -  Where each item is the path name of a
+	pre-compliled stylesheet that you'd like to include.
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/primer.rst]: "compiled"

File: docs/reference/contrib/modeladmin/primer.rst [hunk 21]
```diff
@@ -0,0 +1,397 @@
...
+
+-	``inspect_view_extra_js`` - Where each item is the path name of a javascipt
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/primer.rst]: "javascript"

File: docs/reference/contrib/modeladmin/primer.rst [hunk 22]
```diff
@@ -0,0 +1,397 @@
...
+	file that you'd like to include.
+
+.. _modeladmin_overriding_templates:
+
+--------------------
+Overriding templates
+--------------------
+
+For all modeladmin views, wagtail looks for templates in the following folders
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/primer.rst]: "Wagtail"

File: docs/reference/contrib/modeladmin/primer.rst [hunk 23]
```diff
@@ -0,0 +1,397 @@
...
+-------------------------
+
+While 'view classes' are responsible for a lot of the work, there are also 
+a number of other tasks that ``modeladmin`` must do regularly, that need to be
+handled in a consistent way, and in a number of different places. These tasks
+are designated to set of simple classes (in ``modeladmin``, these are termed 
+'helper' classes) and can be found in ``wagtail.contrib.modeladmin.helpers``.
+
+If you ever intend to write and use your own custom views with ``modeladmin``,
+you should familarise yourself with these helpers, as they are made available
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/primer.rst]: "familiarise"

File: docs/reference/contrib/modeladmin/primer.rst [hunk 24]
```diff
@@ -0,0 +1,397 @@
...
+
+class MyModelAdmin(ModelAdmin):
+	model = MyModel
+	url_helper_class = MyURLHelper
+
+modeladmin_register(MyModelAdmin)
+```
+
+Or, if you have a more complicated use case, where simply setting that 
+attribute isn't possible (due to circlular imports, for example) or doesn't
```
Comment (timorieber) [inline docs/reference/contrib/modeladmin/primer.rst]: "circular"

**Comment by timorieber** [comment]
Great work. Looking forward to use ModelAdmin in my next project.

I had a quick first review with some small suggestions, mostly spelling. Due to my missing experience with ModelAdmin right now I'm not able to review it from a functional perspective.

**Comment by ababic** [comment]
Thanks a lot for casting your eyes over it, @timorieber. There'll be more work to do here in future, for sure. But, I think this is a decent start. I really hope it helps!

**Comment by ababic** [comment]
Hi @kaedroho/@gasman. Just checking this is on your radar? Is there anything else I can do to help move things along?

**Comment by gasman** [comment]
Great work! Sorry for the delay in reviewing this...

Merged in 0c1b67bc1654300a876bf5bebd637394ae8c999a

**Pull request closed without merging**